x-pack/plugins/security_solution/public/timelines/components/flyout/index.tsx

```diff
@@ -30,13 +30,54 @@ interface OwnProps {
   onAppLeave: (handler: AppLeaveHandler) => void;
 }
 
+type VoidFunc = () => void;
+
 const FlyoutComponent: React.FC<OwnProps> = ({ timelineId, onAppLeave }) => {
   const dispatch = useDispatch();
   const getTimelineShowStatus = useMemo(() => getTimelineShowStatusByIdSelector(), []);
   const { activeTab, show, status: timelineStatus, updated } = useDeepEqualSelector((state) =>
     getTimelineShowStatus(state, timelineId)
   );
 
+  const [focusOwnership, setFocusOwnership] = useState(true);
+  const [triggerOnBlur, setTriggerOnBlur] = useState(true);
+  const callbackRef = useRef<VoidFunc | null>(null);
+  const searchRef = useRef<HTMLElement | null>(null);
+
+  const handleSearch = useCallback(() => {
+    if (show && focusOwnership === false) {
+      setFocusOwnership(true);
+    }
+  }, [show, focusOwnership]);
+  const onOutsideClick = useCallback((event) => {
```

Comment:
👍🏾, tested locally and it works. Just curious if it would be simpler to just rely solely on the focus states of the search input? (also renamed `focusOwnership` to `setTimelineFocusOwnership`...I was getting confused at who was owning the focus reading through this 😂 

```
const [isFocusHandlerSet, updateIsFocusHandlerSet] = useState(false);
const removeTimelineFocus = useCallback(() => setTimelineFocusOwnership(false));
const focusOnTimeline = useCallback(() => setTimelineFocusOwnership(true));

const onOutsideClick = useCallback((event) => {
	if (!isFocusHandlerSet) {
		searchRef.current.addEventListener('focus', removeTimelineFocus)
		searchRef.current.addEventListener('blur', focusOnTimeline);
		updateIsFocusHandlerSet(true);
	}	
}

useEffect(() => {
	return () => {
		searchRef.current.removeEventListener('focus', removeTimelineFocus);
		searchRef.current.removeEventListener('blur', focusOnTimeline);
	}
})
```